homeassistant/components/ambient_station/__init__.py

```diff
@@ -58,26 +58,25 @@ def async_hydrate_station_data(data: dict[str, Any]) -> dict[str, Any]:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the Ambient PWS as config entry."""
-    hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN][entry.entry_id] = {}
-
     if not entry.unique_id:
         hass.config_entries.async_update_entry(
             entry, unique_id=entry.data[CONF_APP_KEY]
         )
 
+    ambient = AmbientStation(
+        hass,
+        entry,
+        Websocket(entry.data[CONF_APP_KEY], entry.data[CONF_API_KEY]),
+    )
+
     try:
-        ambient = AmbientStation(
-            hass,
-            entry,
-            Websocket(entry.data[CONF_APP_KEY], entry.data[CONF_API_KEY]),
-        )
-        hass.loop.create_task(ambient.ws_connect())
-        hass.data[DOMAIN][entry.entry_id] = ambient
+        await ambient.ws_connect()
```

Comment:
Will this return when the connection is established or will this block until the connection is done?

Reply:
Good point: this will block in a retry loop until the connection succeeds. Let me think on how to better solve that.